### PR TITLE
Update MDAnalysis and addition of MDAnalysisTests

### PR DESCRIPTION
**Pull request by RMeli**
Reasons for deprecation of older versions:
* https://github.com/MDAnalysis/mdanalysis/issues/4020#issuecomment-1437503739
* There is no explicit dependency on `py-mdanalysis@:1`
    * Only one other package depends on `py-mdanalysis`
* Allows simplification of the package to ease support of future versions

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 1]
```diff
@@ -14,63 +14,129 @@ class PyMdanalysis(PythonPackage):
...
     depends_on("py-tqdm@4.43.0:", when="@1.0.0:", type=("build", "run"))
 
     depends_on("py-joblib", when="@0.16.0:0.20.1", type=("build", "run"))
     depends_on("py-joblib@0.12:", when="@1.0.0:", type=("build", "run"))
 
     depends_on("py-numpy@1.5.0:", when="@:0.15.0", type=("build", "run"))
     depends_on("py-numpy@1.10.4:", when="@0.16.0:0.19.2", type=("build", "run"))
-    depends_on("py-numpy@1.13.3:", when="@0.20.1:", type=("build", "run"))
+    depends_on("py-numpy@1.13.3:", when="@0.20.1:1.0.0", type=("build", "run"))
+    depends_on("py-numpy@1.21.0:", when="@2.4.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion
    depends_on("py-numpy@1.20.0:", when="@2.4.0:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 2]
```diff
@@ -14,63 +14,129 @@ class PyMdanalysis(PythonPackage):
...
+    depends_on("py-scipy@1.0.0:", when="@0.18.0:1.0.0", type=("build", "run"))
+    depends_on("py-scipy@1.5.0:", when="@2.4.0:", type=("build", "run"))
 
     depends_on("py-scikit-learn", when="@0.16.0:+analysis", type=("build", "run"))
     depends_on("py-seaborn", when="+analysis", type=("build", "run"))
 
     depends_on("py-netcdf4@1.0:", when="+amber", type=("build", "run"))
     depends_on("hdf5", when="+amber", type=("run"))
+    depends_on("py-netcdf4@1.0:", when="+extra_formats", type=("build", "run"))
+    depends_on("hdf5", when="+extra_formats", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion
    depends_on("py-h5py@2.10:", when="+extra_formats", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 3]
```diff
@@ -14,63 +14,129 @@ class PyMdanalysis(PythonPackage):
...
+
+    depends_on("py-pytng", when="@2.4.0:+extra_formats", type=("build", "run"))
+    depends_on("py-chemfiles", when="@2.4.0:+extra_formats", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion
    depends_on("py-chemfiles@0.10:", when="@2.4.0:+extra_formats", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 4]
```diff
@@ -14,63 +14,129 @@ class PyMdanalysis(PythonPackage):
...
+    depends_on("py-pyedr", when="@2.4.0:+extra_formats", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion
    depends_on("py-pyedr@0.7.0:", when="@2.4.0:+extra_formats", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 5]
```diff
@@ -14,63 +14,129 @@ class PyMdanalysis(PythonPackage):
...
 
     depends_on("py-scikit-learn", when="@0.16.0:+analysis", type=("build", "run"))
     depends_on("py-seaborn", when="+analysis", type=("build", "run"))
 
     depends_on("py-netcdf4@1.0:", when="+amber", type=("build", "run"))
     depends_on("hdf5", when="+amber", type=("run"))
+    depends_on("py-netcdf4@1.0:", when="+extra_formats", type=("build", "run"))
+    depends_on("hdf5", when="+extra_formats", type=("build", "run"))
+
+    depends_on("py-pytng", when="@2.4.0:+extra_formats", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion
    depends_on("py-pytng@0.2.3:", when="@2.4.0:+extra_formats", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
I prefer to sort dependencies in the order they appear in pyproject.toml (and sort "when" by newest to oldest) because it makes it easier to review and make sure they match, but that's entirely optional, not a requirement.

**Comment by RMeli** [comment]
Thanks for the review @adamjstewart

> I prefer to sort dependencies in the order they appear in pyproject.toml

This is my preference too and tried it before opening this PR. However, it was causing the diffs to be quite large in this case. I can change them once you are happy with the PR?

**Review by adamjstewart — APPROVED** [review]
Should be good now. Don't worry too much about the diffs. I can merge this as is, or you can reorder the deps now, or you can reorder the deps after the next release once v1 and older is removed, up to you.

**Comment by RMeli** [comment]
Ok, in such case I'll do it a part of the major cleanup when removing the deprecated versions. I do plan to keep this package up to date. Thanks!

**Pull request merged**